src/neo/Persistence/MemorySnapshot.cs

```diff
@@ -49,7 +50,8 @@ public void Dispose()
 
         public void Put(byte[] key, byte[] value)
         {
-            writeBatch[key.EnsureNotNull()] = value;
+            if (value is null) throw new ArgumentNullException(nameof(value));
+            writeBatch[key.EnsureNotNull()[..]] = value[..];
```

Comment:
Why `ensurenotnull` here, assign value to an empty key?

```C#
        public static byte[] EnsureNotNull(this byte[] source)
        {
            return source ?? Array.Empty<byte>();
        }
```

what about

```C#
            if (value is null || key is null) throw new ArgumentNullException(nameof(value), nameof(key));
            writeBatch[key[..]] = value[..];
```

Reply:
Removed.